### PR TITLE
GH-10968 Migrate 'Post.GetPostsBatchForIndexing' to Sync by default

### PR DESCRIPTION
**Pull request by lurcio**
#### Summary
Migrates 'Post.GetPostsBatchForIndexing' to Sync by default.

#### Ticket Link
https://github.com/mattermost/mattermost-server/issues/10968

### CODE REVIEW

**Comment by jespino** [comment]
This needs changes in the enterprise repo, so until we create the proper pr it will fail. So @lurcio don't worry about the test failing, and be patient, this PR will take a bit longer to get merged.

File: store/sqlstore/post_store.go [hunk 1]
```diff
@@ -1129,11 +1129,10 @@ func (s *SqlPostStore) GetPostsByIds(postIds []string) store.StoreChannel {
...
-func (s *SqlPostStore) GetPostsBatchForIndexing(startTime int64, endTime int64, limit int) store.StoreChannel {
-	return store.Do(func(result *store.StoreResult) {
-		var posts []*model.PostForIndexing
-		_, err1 := s.GetSearchReplica().Select(&posts,
-			`SELECT
+func (s *SqlPostStore) GetPostsBatchForIndexing(startTime int64, endTime int64, limit int) ([]*model.PostForIndexing, *model.AppError) {
+	var posts []*model.PostForIndexing
+	_, err := s.GetSearchReplica().Select(&posts,
+		`SELECT
 				PostsQuery.*, Channels.TeamId, ParentPosts.CreateAt ParentCreateAt
```
Comment (jespino) [inline store/sqlstore/post_store.go]: Nit: this should have the proper indentation like before the PR. One tab less for each line of the string except for the select line.

File: store/storetest/post_store.go [hunk 2]
```diff
@@ -1976,7 +1976,7 @@ func testPostStoreGetPostsBatchForIndexing(t *testing.T, ss store.Store) {
 	o3.Message = "zz" + model.NewId() + "QQQQQQQQQQ"
 	o3 = (<-ss.Post().Save(o3)).Data.(*model.Post)
 
-	if r := store.Must(ss.Post().GetPostsBatchForIndexing(o1.CreateAt, model.GetMillis()+100000, 100)).([]*model.PostForIndexing); len(r) != 3 {
+	if r, _ := ss.Post().GetPostsBatchForIndexing(o1.CreateAt, model.GetMillis()+100000, 100); len(r) != 3 {
```
Comment (hanzei) [inline store/storetest/post_store.go]: Would you please check the return here?

**Comment by hanzei** [comment]
@lieut-data Would you also help out with this changes?

File: store/sqlstore/post_store.go (hunk 1 above)
Comment (lurcio) [reply store/sqlstore/post_store.go]: Thank you for the feedback @jespino, I have updated my branch to fix the indentation. Well, I hope I have anyway

File: store/storetest/post_store.go (hunk 2 above)
Comment (lurcio) [reply store/storetest/post_store.go]: Thank you for your feedback @hanzei, I have updated my PR to check the error return value here. I think that's what you wanted, but if not please let me know!

**Comment by lieut-data** [comment]
@hanzei, I've pushed up https://github.com/mattermost/enterprise/pull/448, but I think Jenkins isn't currently smart enough to let either build pass since there's no way for it to know to build one with the other. `go vet` passes locally, so I propose we just merge both together once the `enterprise` code has been reviewed. (@grundleborg and @jespino are assigned.)

Also am proposing that -- going forward -- we avoid help wanted tickets for SQLStore changes used by the enterprise code for exactly this difficulty.

**Comment by hanzei** [comment]
@lieut-data Thanks for finishing the EE changes.

> Also am proposing that -- going forward -- we avoid help wanted tickets for SQLStore changes used by the enterprise code for exactly this difficulty.

100% agree. I added a note about this to the docs: https://github.com/mattermost/docs/pull/2771

**Pull request merged**